### ISSUE TITLE
Integrate OpenJMH for per-binding microbenchmarks

### ISSUE THREAD

**Issue by jfallows** [aklivity/zilla#167]
Today we have an `OpenJMH` microbenchmark [`BufferBM`](https://github.com/aklivity/zilla/blob/develop/runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/internal/concurrent/bench/BufferBM.java) for the internal ring buffer data structure used by all zilla bindings.

However, this does not measure the code executed in reaction to reading a frame from the ring buffer, such as the dispatch logic in the runtime engine, nor any binding specific code needed to implement a specific protocol decoder for example.

Therefore, we need to provide an integration with `OpenJMH` that will make it straightforward to write microbenchmarks for each binding, driving frames into the ring buffer (and consuming any reply frames), measuring the performance of each binding in isolation.

As we make performance improvements to the codebase, we should see these benchmarks reflect the improvements, perhaps for a specific binding, or for all bindings if the improvement is made generically in the engine.

In a similar fashion to the integration with `k3po`, we can implement the integration with `OpenJMH` in `engine` `test` codebase, then reuse that from each binding to run binding-specific microbenchmarks.

**Comment by Alfusainey** [comment]
@jfallows 

> In a similar fashion to the integration with k3po, we can implement the integration with OpenJMH in engine test codebase, then reuse that from each binding to run binding-specific microbenchmarks.

In addition to k3po, can we also say that its a (remotely) similar fashion to the idea behind the `EngineRule` test rule (found in `engine` `test`)? I see that the EngineRule is used in various binding and guard-specific integration tests but it is defined in the engine test code base

**Comment by jfallows** [comment]
@Alfusainey yes that's right, even though the execution environment for JMH micro benchmarks does not use JUnit.
